src/tests/testSBuf.cc

```diff
@@ -1190,6 +1193,23 @@ TestSBuf::testSBufHash()
     }
 }
 
+void
+TestSBuf::testStdAlgorithm()
+{
+    {
+        SBuf src("foo"), dst;
+        std::copy(src.begin(), src.end(), std::back_inserter(dst));
+        CPPUNIT_ASSERT_EQUAL(dst, src);
+    }
+
+    {
+        SBuf src("foo"), dst;
+        std::copy_if(src.begin(), src.end(), std::back_inserter(dst),
+            [](char c) { return c != 'o'; });
+        CPPUNIT_ASSERT_EQUAL(dst, SBuf("f"));
```

Comment:
Please make this test a little more clear that the string is not just being mangled badly.
```suggestion
        SBuf src("foaro"), dst;
        std::copy_if(src.begin(), src.end(), std::back_inserter(dst),
            [](char c) { return c != 'o'; });
        CPPUNIT_ASSERT_EQUAL(dst, SBuf("far"));
```

Reply:
I changed to use the standard string used throughout the test, this should offer a greater confidence in the result